src/Bicep.Core/LanguageConstants.cs

```diff
@@ -115,6 +121,26 @@ public static IEnumerable<TypeProperty> GetCommonResourceProperties(ResourceType
             yield return new TypeProperty("apiVersion", new StringLiteralType(reference.ApiVersion), TypePropertyFlags.ReadOnly | TypePropertyFlags.SkipInlining);
         }
 
+        public static TypeSymbol CreateModuleType(IEnumerable<TypeProperty> paramsProperties, IEnumerable<TypeProperty> outputProperties, string typeName)
+        {
+            var paramsType = new NamedObjectType(ModuleParamsPropertyName, TypeSymbolValidationFlags.Default, paramsProperties, null);
+            // If none of the params are reqired, we can allow the 'params' declaration to be ommitted entirely
```

Comment:
I like that our type system is rich enough to allow us to express stuff like this!